dql/history.py

```diff
@@ -0,0 +1,49 @@
+import os
+from pathlib import Path
+
+
+class HistoryManager(object):
+    _initial_history_length = 0
+
+    def _create_dir_if_not_exists(self, path: str) -> None:
+        if not os.path.exists(path):
+            os.makedirs(path)
+
+    def _create_file_if_nost_exists(self, path: str) -> None:
+        if not os.path.exists(path):
+            with open(path, "w"):
+                pass
+
+    def _get_history_dir_and_file(self):
+        home = str(Path.home())
+        history_dir = os.path.join(home, ".dql")
+        self._create_dir_if_not_exists(history_dir)
+        history_file = os.path.join(history_dir, "history")
+        self._create_file_if_nost_exists(history_file)
+        return (history_dir, history_file)
+
+    def try_to_load_history(self):
+        (history_dir, history_file) = self._get_history_dir_and_file()
+        if os.path.exists(history_file):
+            # print("History loading from file: " + history_file)
+            try:
+                import readline
+            except ImportError:
+                # Windows doesn't have readline, so gracefully ignore.
+                pass
+            else:
+                readline.read_history_file(history_file)
+                self._initial_history_length = readline.get_current_history_length()
+
+    def try_to_write_history(self):
+        (history_dir, history_file) = self._get_history_dir_and_file()
```

Comment:
Here you can just add:
```python
os.makedirs(os.path.dirname(history_file), exist_ok=True)
```
Presumably the file itself doesn't have to exist before you call `append_history_file`

Reply:
`append_history_file` needs the file to exist.

$> python 
Interpreter
```python
import readline
readline.append_history_file(1, '/Users/sai/.dql/history_1')
```
will throw. 
```
FileNotFoundError: [Errno 2] No such file or directory
```

Reply:
😧